hcmanager/utils_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcmanager
+
+import (
+	"reflect"
+
+	hwcc "github.com/metal3-io/hardware-classification-controller/api/v1alpha1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	bmoapis "github.com/metal3-io/baremetal-operator/pkg/apis"
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/klogr"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("HCManager", func() {
+
+	type testCaseFetchBMH struct {
+		namespace      string
+		expectedError  bool
+		expectedResult []bmh.BareMetalHost
+	}
+
+	DescribeTable("Test fetch BaremetalHost list",
+		func(tc testCaseFetchBMH) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), getHosts()...)
+			hcManager := NewHardwareClassificationManager(c, klogr.New())
+			result, _, err := hcManager.FetchBmhHostList(tc.namespace)
+			if tc.expectedError {
+				Expect(err).To(HaveOccurred())
+			} else if len(tc.expectedResult) == 0 {
+				Expect(len(result)).To(BeZero())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(result)).Should(Equal(len(tc.expectedResult)))
+				Expect(result).Should(Equal(tc.expectedResult))
+				Expect(reflect.DeepEqual(result, tc.expectedResult)).To(BeTrue())
```

Comment:
Expect(result).Should(Equal(tc.expectedResult)) is sufficient here. it does a deepequal. and if the deepequal is true, then the length of both must be the same.

Reply:
done